homeassistant/components/climate/__init__.py

```diff
@@ -656,11 +720,31 @@ async def async_turn_off(self) -> None:
         if HVACMode.OFF in self.hvac_modes:
             await self.async_set_hvac_mode(HVACMode.OFF)
 
-    @cached_property
+    @property
     def supported_features(self) -> ClimateEntityFeature:
         """Return the list of supported features."""
         return self._attr_supported_features
 
+    @supported_features.setter
+    def supported_features(self, new_features: ClimateEntityFeature) -> None:
+        """Return the list of supported features.
+
+        For backwards compatibility for turn_on/turn_off
+        ClimateEntityFeatures
+        """
+        self.__mod_supported_features = new_features
+
+    @supported_features.getter
+    def supported_features(self) -> ClimateEntityFeature:
+        """Return the list of supported features.
+
+        For backwards compatibility for turn_on/turn_off
+        ClimateEntityFeatures
+        """
+        if mod_supported_features := self.__mod_supported_features:
+            return mod_supported_features
+        return ClimateEntityFeature(0)
+
```

Comment:
As discussed on Discord, implementing a `__getattribute__` with a special treatment should have a higher chance of success.

Maybe something like this will work:
```py
def ___getattribute__(self, name):
    if name != "supported_features":
        return super().___getattribute__(name)
    return super().___getattribute__(name) | super().___getattribute__("_CimateEntity__mod_supported_features")
```